src/libraries/System.Security.Cryptography/tests/OpenSslNamedKeysTests.manual.cs

```diff
@@ -0,0 +1,194 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Tests
+{
+    // See osslplugins/README.md for instructions on how to build and install the test engine and setup for TPM tests.
+    [SkipOnPlatform(~TestPlatforms.Linux, "Named keys tests are only supported on Linux")]
+    public class OpenSslNamedKeysTests
+    {
+        const string EnvVarPrefix = "DOTNET_CRYPTOGRAPHY_TESTS_ENGINE_";
+        const string TestEngineEnabledEnvVarName = EnvVarPrefix + "ENABLE";
+        const string TestEngineEnsureFailingEnvVarName = EnvVarPrefix + "ENSURE_FAILING";
+        const string TpmTssEngineEcDsaKeyHandleEnvVarName = EnvVarPrefix + "TPM_ECDSA_KEY_HANDLE";
+
+        const string NonExistingEngineName = "dntestnonexisting";
+        const string NonExistingEngineKeyName = "nonexisting";
+
+        const string TestEngineName = "dntest";
+        const string TestEngineKeyId = "first";
+        const string TpmTssEngineName = "tpm2tss";
+
+        public static bool ShouldRunEngineTests { get; private set; }
+        public static bool ShouldFailTests { get; private set; }
+        public static bool ShouldRunTpmTssTests { get; private set; }
+
+        public static string TpmTssEngineEcDsaKeyHandle { get; private set; }
+
+        static OpenSslNamedKeysTests()
+        {
+            TpmTssEngineEcDsaKeyHandle = Environment.GetEnvironmentVariable(TpmTssEngineEcDsaKeyHandleEnvVarName);
+            ShouldRunEngineTests = StringToBool(Environment.GetEnvironmentVariable(TestEngineEnabledEnvVarName));
+            ShouldFailTests = StringToBool(Environment.GetEnvironmentVariable(TestEngineEnsureFailingEnvVarName));
+
+            ShouldRunTpmTssTests = !string.IsNullOrEmpty(TpmTssEngineEcDsaKeyHandle);
+        }
```

Comment:
So long as `TpmTssEngineEcDsaKeyHandle`s declaration is moved above the three booleans, all of these initializer values could be done inline.  Why create an explicit cctor?

```C#
public static bool ShouldRunEngineTests { get; } = StringToBool(Environment.GetEnvironmentVariable(TestEngineEnabledEnvVarName));
```